src/strands/strands_glslBackend.js

```diff
@@ -0,0 +1,205 @@
+import { NodeType, OpCodeToSymbol, BlockType, OpCode, NodeTypeToName, isStructType, StructType, StatementType } from "./ir_types";
+import { getNodeDataFromID, extractNodeTypeInfo } from "./ir_dag";
+import * as FES from './strands_FES'
+
+function shouldCreateTemp(dag, nodeID) {
+  const nodeType = dag.nodeTypes[nodeID];
+  if (nodeType !== NodeType.OPERATION) return false;
+  const uses = dag.usedBy[nodeID] || [];
+  return uses.length > 1;
+}
+
+const TypeNames = {
+  'float1': 'float',
+  'float2': 'vec2',
+  'float3': 'vec3',
+  'float4': 'vec4',
+
+  'int1':   'int',
+  'int2':   'ivec2',
+  'int3':   'ivec3',
+  'int4':   'ivec4',
+
+  'bool1':  'bool',
+  'bool2':  'bvec2',
+  'bool3':  'bvec3',
+  'bool4':  'bvec4',
+
+  'mat2':   'mat2x2',
+  'mat3':   'mat3x3',
+  'mat4':   'mat4x4',
+}
+
+const cfgHandlers = {
+  [BlockType.DEFAULT]: (blockID, strandsContext, generationContext) => {
+    const { dag, cfg } = strandsContext;
+
+    const instructions = cfg.blockInstructions[blockID] || [];
```

Comment:
Do these have to be sorted by DAG order to be valid? (Are these naturally stored in sorted order already?)